# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1367564263 14400
#      Fri May 03 02:57:43 2013 -0400
# Node ID 6698d0c42fbf576e8898642c2b03933e99f81116
# Parent  2f8fb02a6c311a05a9a4a23fe45d0322d880ab31
unix terminal: don't blink cursor if command window does not have focus

* TerminalView.cpp (TerminalView::drawCursor): Always draw cursor if
window does not have focus.

diff --git a/libgui/qterminal/libqterminal/unix/TerminalView.cpp b/libgui/qterminal/libqterminal/unix/TerminalView.cpp
--- a/libgui/qterminal/libqterminal/unix/TerminalView.cpp
+++ b/libgui/qterminal/libqterminal/unix/TerminalView.cpp
@@ -525,17 +525,17 @@ void TerminalView::drawCursor(QPainter& 
                               const QRect& rect,
                               const QColor& foregroundColor,
                               const QColor& /*backgroundColor*/,
                               bool& invertCharacterColor)
 {
   QRect cursorRect = rect;
   cursorRect.setHeight(_fontHeight - _lineSpacing - 1);
 
-  if (!_cursorBlinking)
+  if (!_cursorBlinking || !hasFocus())
     {
       if ( _cursorColor.isValid() )
         painter.setPen(_cursorColor);
       else {
           painter.setPen(foregroundColor);
         }
 
       if ( _cursorShape == BlockCursor )
